### ISSUE TITLE
Please tag a versioned release

### ISSUE THREAD

**Issue by hishamhm** [daurnimator/ldbus#5]
This rock is at the bottom of a dependency chain for the Awesome WM PulseAudio widget. :)

Also, for a more straightforward installation, wouldn't it be nicer to move the `dbus-1.0/` part of the incdir to the #include calls? (And also testing for `"dbus-1.0/dbus/dbus.h"` in the rockspec instead). It would make `luarocks install dbus` work right out of the box once a release is made, and it would encode the desired API version in the source code, which I believe is a nice thing for compatibility.

**Comment by daurnimator** [comment]
> Also, for a more straightforward installation, wouldn't it be nicer to move the dbus-1.0/ part of the incdir to the #include calls? (And also testing for "dbus-1.0/dbus/dbus.h" in the rockspec instead). It would make luarocks install dbus work right out of the box once a release is made, and it would encode the desired API version in the source code, which I believe is a nice thing for compatibility.

Note that `DBUS_ARCH_INCDIR` is usually in /usr/lib not /usr/include.

**Comment by hishamhm** [comment]
> Note that DBUS_ARCH_INCDIR is usually in /usr/lib not /usr/include.

Oh, I clearly missed it. In any case, the versioned release helps. :)

As for the location of include dirs, the best way to improve this would be for LuaRocks to support pkgconfig on Unix.

**Comment by daurnimator** [comment]
> the best way to improve this would be for LuaRocks to support pkgconfig on Unix.

I'm not sure about pkgconfig support. But what about being able to provide some "default" search locations? e.g. in `external_dependencies` I could suggest searching in /usr/lib/dbus-1.0.
This would help with several other projects too.

**Comment by hishamhm** [comment]
> But what about being able to provide some "default" search locations?

Ah, that could work too! I think both this approach and pkgconfig have complementary pros and cons. Perhaps we could add both.

**Comment by daurnimator** [comment]
Reviewing my own code I found two significant bugs, one fixed in HEAD (4e95fff) and one now filed as #6. The latter will require a significant refactor; something I don't have time for.

Do you still want a release?

**Comment by hishamhm** [comment]
> Do you still want a release?

I ended up going through a different approach here, so I won't be using this immediately. I'll ping you again if it proves necessary again. But thanks in any case!

**Comment by stefano-m** [comment]
Hi, I *think* @hishamhm is referring to my [Awesome WM Widget](https://github.com/stefano-m/awesome-pulseaudio_widget) that uses ldbus via a [higher level api (also written by me)](https://github.com/stefano-m/lua-ldbus_api).

I have not bothered you since I saw that development on ldbus had been still for a while, but it would actually be great if you managed to pin a release at your convenience so I can pin my dependencies too. No hurry though, and thank you for writing ldbus!

**Comment by hishamhm** [comment]
@stefano-m That's right! For my use case, I ended up going with an ALSA widget for now but I plan to revisit yours in the future. :)

**Comment by daurnimator** [comment]
FWIW I use [pasystray](https://github.com/christophgysin/pasystray) what does using an awesome widget do over that? (I can use mouse wheel to scroll over it to change volume; I have bindings from volume up/down keys to just run `pactl`)

**Comment by stefano-m** [comment]
> What does using an awesome widget do over that?

That's a bit OT, bu whatever :smile: So, I have started to use the Awesome WM and was unhappy with the widgets I found around: almost all of them use timers and parse text output from command line using `io.popen` or similar. I wanted to leverage DBus (asynchronous API, and Awesome has also good built-in support for signals) and also have nice icons and took the opportunity to learn some Lua in the meantime. I always thought I'd be the only user of my code.

Regarding this issue. As I said, it would be great if you could tag a version if/when you have time. Of course if my widgets gain any more traction, your library may get more attention as well!

BTW, thanks again for writing `ldbus`, I wouldn't have been able to write my widgets without it!

**Comment by stefano-m** [comment]
> https://github.com/daurnimator/ldbus/issues/5#issuecomment-251259000

@hishamhm FYI I have heavily refactored https://github.com/stefano-m/awesome-pulseaudio_widget and have replaced ldbus with [lgi](https://github.com/pavouk/lgi/). The catch is that it will work on **Awesome 4.x** only though.

I still have one of my widgets that depend on ldbus, but I hope I will move it to lgi eventually.

ldbus was great to learn DBus and co, but I appreciate that daurnimator is not working actively on it. On the other hand, lgi is actively developed and is also a dependency of Awesome.

(sorry for the OT comment)

**Comment by hishamhm** [comment]
@stefano-m Thanks for the heads up! :)
